src/framework/filterControl/FilterControl.jsx

```diff
@@ -37,108 +50,130 @@ export default class FilterControl extends Component {
+      !this.formDropdown.contains(event.target)
+    ) {
+      this.setState({
+        selectedFilterOption: null,
+      });
+    }
+  }
+
   onDropdownClose() {
     this.setState({
       isDropdownOpen: false,
       selectedFilterOption: null,
     });
   }
 
-  onSelectFilterOption(filter) {
+  onSelectFilterOption(filterOption) {
     this.setState({
-      isDropdownOpen: true,
-      selectedFilterOption: filter,
+      selectedFilterOption: filterOption,
     });
   }
 
-  onToggleClick() {
+  onAddButtonClick() {
     this.setState({
-      isDropdownOpen: !this.state.isDropdownOpen,
+      isDropdownOpen: true,
       selectedFilterOption: null,
     });
   }
 
   renderDropdownContent() {
-    // If we don't have a selected filter yet, then we're in the process
-    // of selecting one.
-    if (!this.state.selectedFilterOption) {
-      return (
-        <div>
-          <div className="filterDropdown__header">
-            <div className="filterDropdown__header__title">
-              Add a Filter
-            </div>
-            <span
-              className="filterDropdown__header__closeButton"
-              onClick={this.onDropdownClose}
-            />
-          </div>
-          <FilterOptionList
-            filterOptions={this.props.filterOptions}
-            onSelectFilterOption={this.onSelectFilterOption}
-          />
-        </div>
-      );
-    }
-
-    // If we have a selected filter, then we can create a filter matcher
-    // from it.
     return (
       <div>
         <div className="filterDropdown__header">
-          <span
-            className="filterDropdown__header__backButton"
-            onClick={this.onCancelFilter}
-          />
           <div className="filterDropdown__header__title">
-            {this.state.selectedFilterOption.name}
+            Add a Filter
           </div>
+          <span
+            className="filterDropdown__header__closeButton"
+            onClick={this.onDropdownClose}
+          />
         </div>
-        <FilterForm
+        <FilterOptionList
+          filterOptions={this.props.filterOptions}
+          onSelectFilterOption={this.onSelectFilterOption}
+        />
+      </div>
+    );
+  }
+
+  renderFilterItems() {
+    const filterItems = this.props.selectedFilters.map(filter => (
+      <FilterItem
+        filter={filter}
+        key={filter.filterOption.name}
+        onRemoveSelectedFilter={this.props.onRemoveSelectedFilter}
+        onReplaceFilter={this.props.onReplaceFilter}
+      />
+    ));
+
+    let formDropdown;
+    if (this.state.selectedFilterOption !== null) {
+      formDropdown = (
+        <FilterFormDropdown
           filterOption={this.state.selectedFilterOption}
           onAddFilter={this.onAddFilter}
+          onBackButtonClick={this.onBackButtonClick}
         />
+      );
+    }
+
+    return (
+      <div className="filterItemList">
+        {filterItems}
+        <div ref={(div) => { this.formDropdown = div; }}>
```

Comment:
It does not work this way, because of http://eslint.org/docs/rules/no-return-assign.